hackernews/program.py

```diff
@@ -10,18 +10,20 @@
 
 slack = slack_client("slack_connection")
 SLACK_CHANNEL = os.getenv("SLACK_CHANNEL")
+API_URL = "http://hn.algolia.com/api/v1/search_by_date?tags=story&page=0&query="
 
 
 def on_slack_command(event):
     """
     Workflow's entry-point.
 
-    The function extracts a topic from a Slack command, monitors for new articles,
+    extract a topic from a Slack command, monitors for new articles,
     and posts updates to the Slack channel.
     """
-    topic = event.data.text
+    topic = event.data.text.split(" ", 1)[1].strip()
```

Comment:
it is not guaranteed you'll have a two items in the returned list, so `[1]` might throw.

Reply:
Actually it is guaranteed because the text of a slack command is "/slack-command optional parameters..." so at least 1 space has to exist. But I agree that `[-1]` does the same and is safer just in case.

Reply:
I’ll get an error if there’s no topic after the app mention, but it can be fixed with:
parts = event.data.text.split(" ", 1)
if len(parts) > 1: topic = parts[1].strip()
However, I don’t think it’s necessary.

Reply:
No `[-1]` means "last element" - whether or not it's the first and only element in the list.